node/node.go

```diff
@@ -631,27 +634,27 @@ func (node *Node) UpdateStakingList(block *types.Block) error {
 	txns := block.Transactions()
 	for i := range txns {
 		txn := txns[i]
-		value := txn.Value().Int64()
-		currentSender, _ := types.Sender(signerType, txn)
-		_, isPresent := node.CurrentStakes[currentSender]
 		toAddress := txn.To()
 		if *toAddress != node.StakingContractAddress { //Not a address aimed at the staking contract.
 			continue
 		}
-		//This should be based on a switch case on function signature.
-		//TODO (ak) https://github.com/harmony-one/harmony/issues/430
-		if value > int64(0) { //If value >0 means its a staking deposit transaction
+		currentSender, _ := types.Sender(signerType, txn)
+		_, isPresent := node.CurrentStakes[currentSender]
+		data := txn.Data()
+		switch funcSignature := decodeFuncSign(data); funcSignature {
+		case "0xd0e30db0": //deposit
```

Comment:
constant.

I made easy for you to check in the code a few times but i could not let it happen again.